src/fmeval/eval_algorithms/faithfulness.py

```diff
@@ -0,0 +1,297 @@
+            output_keys=[output_key],
+        )
+        self.input_key = input_key
+        self.output_key = output_key
+        self.judge_model = judge_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Invokes the judge model to generate statements and arguments the input record with the statements.
+        :param record: The input record.
+        :returns: The input record with the mean added in.
+        """
+        get_raw_statements = GetModelOutputs(
+            input_to_output_keys={self.input_key: ["raw_statements"]},
+            model_runner=self.judge_model,
+        )
+        record_with_raw_statements = get_raw_statements(record)
+        raw_statements = record_with_raw_statements["raw_statements"].split("\n")
+        list_statements = [statement for statement in raw_statements if statement.startswith("Statement:")]
+        statements_str: str = "\n".join([f"{i + 1}.{st}" for i, st in enumerate(list_statements)])
+        record[self.output_key] = statements_str
+        return record
+
+
+class Faithfulness(EvalAlgorithmInterface):
+    """ """
+
+    eval_name = EvalAlgorithm.FAITHFULNESS.value
+
+    def __init__(self):
+        """Faithfulness initializer."""
+        super().__init__(EvalAlgorithmConfig())
+
+    @staticmethod
+    def _build_pipeline(
+        judge_model: ModelRunner,
+        long_form_prompt_template: str,
+        nli_statements_prompt_template: str,
+    ) -> TransformPipeline:
+        gen_long_form_prompt = GeneratePrompt(
+            input_keys=[],
+            output_keys=["long_form_prompt"],
+            prompt_template=long_form_prompt_template,
+            placeholder_to_record_key={
+                "question": DatasetColumns.MODEL_INPUT.value.name,
+                "answer": DatasetColumns.MODEL_OUTPUT.value.name,
+            },
+        )
+        get_statements = GetStatements(
+            input_key="long_form_prompt",
+            output_key="statements",
+            judge_model=judge_model,
+        )
+        gen_nli_statements_prompt = GeneratePrompt(
+            input_keys=[],
+            output_keys=["nli_statements_prompt"],
+            prompt_template=nli_statements_prompt_template,
+            placeholder_to_record_key={"context": DatasetColumns.TARGET_CONTEXT.value.name, "statements": "statements"},
+        )
+        get_raw_verdicts = GetModelOutputs(
+            input_to_output_keys={"nli_statements_prompt": ["raw_verdicts"]},
+            model_runner=judge_model,
+        )
+        compute_score = FaithfulnessScore()
+        transforms = [gen_long_form_prompt, get_statements, gen_nli_statements_prompt, get_raw_verdicts, compute_score]
+        pipeline = TransformPipeline(transforms)
+        return pipeline
+
+    def evaluate_sample(
+        self,
+        model_input: str,
+        model_output: str,
+        target_context: str,
+        judge_model: Optional[ModelRunner] = None,
```

Comment:
Could `judge_model`, `long_form_prompt_template` and  `nli_statements_prompt_template` be a variable of the evaluation algorithm itself (pass it in `__init__()`) instead of adding them to `evaluate` and `evaluate_sample` functions? 

Is there any reason it has to be a parameter of the functions?